adagrams/game.py

```diff
@@ -1,13 +1,105 @@
 from random import randint
 
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
 def draw_letters():
-    pass
+
+    pool = []
+    for letter, count in LETTER_POOL.items():
+        for _ in range(count):
+            pool.append(letter)
+    
+    letter_bank = []
+    while len(letter_bank) < 10:
+        index = randint(0, len(pool) - 1)
+        letter = pool[index]
+        letter_bank.append(letter)
+        pool.pop(index) 
+
+    return letter_bank
 
 def uses_available_letters(word, letter_bank):
-    pass
+    
+    word = word.upper()
+    letter_bank_counts = {}
+
+    for letter in letter_bank:
+        letter = letter.upper()
+        if letter in letter_bank_counts:
+            letter_bank_counts[letter] += 1
+        else:
+            letter_bank_counts[letter] = 1
```

Comment:
A more concise and typical way you'll see this written in Python is like:

```suggestion
    for letter in letter_bank:
            letter = letter.upper()
            letter_bank_counts[letter] = letter_bank_counts.get(letter, 0) + 1
```